tests/conftest.py

```diff
@@ -40,16 +41,47 @@
 from helpers import create_record
 from invenio_db import db as db_
 from invenio_db import InvenioDB
+from invenio_indexer import InvenioIndexer
 from invenio_indexer.api import RecordIndexer
 from invenio_pidstore import InvenioPIDStore
 from invenio_records import InvenioRecords
+from invenio_records.api import Record
 from invenio_rest import InvenioREST
 from invenio_search import InvenioSearch, RecordsSearch, current_search, \
     current_search_client
 from sqlalchemy_utils.functions import create_database, database_exists
 
 from invenio_records_rest import InvenioRecordsREST, config
 from invenio_records_rest.facets import terms_filter
+from invenio_records_rest.utils import allow_all, deny_all
+
+
+def deny_record(*args, **kwargs):
+    """Return permission that always deny an access."""
+    if 'record' in kwargs and kwargs['record']:
+        return type('Deny', (), {'can': lambda self: False})()
+    else:
+        return type('Allow', (), {'can': lambda self: True})()
+
+
+def deny_login(*args, **kwargs):
+    """Return permission that deny access simulating required_login."""
+    if 'record' in kwargs and kwargs['record']:
+        return type('Deny', (), {'can': lambda self: True})()
+    else:
+        return type('Allow', (), {'can': lambda self: False})()
+
+
+class IndexingRecord(Record):
```

Comment:
consider using `indexer` fixture explicitly instead of overriding `create` method.